test/middleware/spec.check-progress.js

```diff
@@ -50,4 +51,164 @@ describe('middleware/check-session', function () {
+
+            req.sessionModel.set(sessionFields);
+            req.sessionModel.set({
+                steps: sessionSteps
+            });
+
+            controller.options.next = steps['/step1'].next;
+            controller.options.forks = steps['/step1'].forks;
+
+            checkProgress('/step1', controller, steps, '/');
+        });
+
+        describe('when all steps are still reachable after forking', function () {
+            beforeEach(function () {
+                controller.getNextStep = sinon.stub().returns('/fork1');
+                controller.emit('complete', req, res);
+            });
+
+            it('no steps are removed', function () {
+                req.sessionModel.get('steps').should.be.eql(sessionSteps)
+            });
+
+            it('doesn\'t invalidate any fields', function () {
+                req.sessionModel.toJSON().should.contain.all.keys(
+                    'step-1-field-1',
+                    'step-1-field-2',
+                    'step-2-field-1',
+                    'step-2-field-2',
+                    'step-3-field-1',
+                    'step-3-field-2',
+                    'step-4-field-1',
+                    'step-4-field-2',
+                    'fork-field-1',
+                    'fork-field-2'
+                );
+            });
+        });
+
+        describe('When result of forking skips a single step', function () {
+
+            beforeEach(function () {
+              controller.getNextStep = sinon.stub().returns('/step2');
+              controller.emit('complete', req, res);
+            });
+
+            it('invalidates fork1', function () {
+                req.sessionModel.get('steps').should.not.contain('/fork1');
+            });
+
+            it('invalidates fork-field-1 and fork-field-2', function () {
+                expect(req.sessionModel.get('fork-field-1')).to.be.undefined;
+                expect(req.sessionModel.get('fork-field-2')).to.be.undefined;
+            });
+
+            it('doens\'t invalidate step2', function () {
+                req.sessionModel.get('steps').should.contain('/step2');
+            });
+
+            it('doens\'t remove step-2-field-1 or step-2-field-2', function () {
+                req.sessionModel.get('step-2-field-1').should.be.true;
+                req.sessionModel.get('step-2-field-2').should.be.true;
+            });
+
+        });
+
+        describe('when the result of forking skips multiple steps', function () {
+
+            beforeEach(function () {
+                controller.getNextStep = sinon.stub().returns('/step4');
+                controller.emit('complete', req, res);
+            });
+
+            it('invalidates step2, step3, and fork1', function () {
+                req.sessionModel.get('steps').should.not.contain.any('/step2', '/step3', '/fork1');
+            });
+
+            it('doesn\'t invalidate step1 or step4', function () {
```

Comment:
doesn't invalidate steps in current journey? Or maybe new journey depending on how you look at it.